cime_config/hist_config.py

```diff
@@ -0,0 +1,1126 @@
+        return add_ok
+
+    def add_max_fields(self, fields, pobj, logger):
+        """Add one or more max_fields to this HistoryVolConfig object.
+        Return True if it was okay to add <fields> to list of max fields.
+        """
+        add_ok = self.__max_fields.add_fields(fields, pobj, logger)
+        return add_ok
+
+    def add_var_fields(self, fields, pobj, logger):
+        """Add one or more var_fields to this HistoryVolConfig object.
+        Return True if it was okay to add <fields> to list of var fields.
+        """
+        add_ok = self.__var_fields.add_fields(fields, pobj, logger)
+        return add_ok
+
+    def remove_fields(self, fields, pobj, logger):
+        """Remove each field in <fields> from all lists it is on.
+        Return True if each field was found (and removed)."""
+        fields_to_delete = set(fields)
+        fields_deleted = set()
+        all_removed = True
+        for fld_list in self.__all_fields:
+            removed = fld_list.remove_fields(fields_to_delete, pobj, logger)
+            fields_deleted.update(removed)
+        # end for
+        if fields_to_delete != fields_deleted:
+            all_removed = False
+            missing_fields = fields_to_delete.difference(fields_deleted)
+            ctx = context_string(pobj)
+            logger.warning(self.__missing_msg.format(", ".join(list(missing_fields)),
+                                       self.volume, ctx))
+        # end if
+        return all_removed
+
+    @property
+    def volume(self):
+        """Return the volume for this HistoryVolConfig object"""
+        return self.__volume
+
+    @property
+    def precision(self):
+        """Return the precision property for this HistoryVolConfig object"""
+        return self.__precision
+
+    def set_precision(self, prec, pobj, logger):
+        """Modify the precision property of this HistoryVolConfig object.
+        Return True if <prec> is a recognized precision"""
+        if prec.upper() in _OUT_PRECS:
+            self.__precision = prec
+            self.__precision_set = True
+            if logger.getEffectiveLevel() <= logging.DEBUG:
+                ctx = context_string(pobj)
+                logger.debug(f"Setting precision to '{prec}'{ctx}")
+            # end if
+            return True
+        # end if
+        emsg = f"Attempt to set unrecognized precision, '{prec}'"
+        pobj.add_syntax_err(emsg)
+        return False
+
+    @property
+    def max_frames(self):
+        """Return the max_frames property for this HistoryVolConfig object"""
+        return self.__max_frames
+
+    @property
+    def write_nstep0(self):
+        """Return the write_nstep0 property for this HistoryVolConfig object"""
+        return self.__write_nstep0
+
+    def set_max_frames(self, nframes, pobj, logger):
+        """Modify the max_frames property of this HistoryVolConfig object.
+        Return True if <nframes> is a valid setting."""
```

Comment:
```suggestion
        Return True if <nframes> is a valid setting and max_frames is updated.
```
Might want to clarify condition.

Reply:
updated.